### PR TITLE
feat: changed topbar balance view for futures

### PR DESCRIPTION
**Pull request by muradmalik777**

### CODE REVIEW

File: src/components/Balance/History/FuturesHistoryTable.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+  /**
+   * @type {MUIDataTableOptions}
+   */
+  const options = {
+    sortOrder: { name: "date", direction: "desc" },
+  };
+
+  const dynamicColumns = () => {
+    for (let a = 0; a < quotes.length; a++) {
```
Comment (cwagner22) [inline src/components/Balance/History/FuturesHistoryTable.js:116]: Can we use `i` instead? ;)

File: src/components/Balance/ProfitLossAnalysis/ProfitLossAnalysis.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  };
+
+  const [profitLossData, setProfitLossData] = useState(inittialValues);
+
+  const prepareData = () => {
+    let data = { ...inittialValues };
+    const currentEquity = dailyBalance.balances.length
+      ? dailyBalance.balances[dailyBalance.balances.length - 1]
+      : createEmptyUserEquityEntity();
+    [...dailyBalance.balances].forEach((item) => {
```
Comment (cwagner22) [inline src/components/Balance/ProfitLossAnalysis/ProfitLossAnalysis.js:36]: No need destructuring here, `dailyBalance.balances.forEach` is fine.

File: src/components/Balance/TotalEquity/TotalEquityGraph/TotalEquityGraph.js [hunk 3]
```diff
@@ -40,8 +42,9 @@ const TotalEquityGraph = ({ list, modal }) => {
   };
 
   const prepareChartData = () => {
+    const key = selectedExchange.exchangeType === "futures" ? "totalWalletUSDT" : "totalUSDT";
     [...list].forEach((item) => {
```
Comment (cwagner22) [inline src/components/Balance/TotalEquity/TotalEquityGraph/TotalEquityGraph.js:46]: `list.forEach`

File: src/components/Balance/TotalEquity/TotalEquityGraph/TotalEquityGraph.js [hunk 4]
```diff
@@ -60,6 +63,29 @@ const TotalEquityGraph = ({ list, modal }) => {
...
+      return (
+        <Box className="equityTooltip">
+          <Box>
+            <span className="label"> Date:</span>
+            <span>{list[tooltipItem.index] ? list[tooltipItem.index].date : "0"}</span>
+          </Box>
+          {list[tooltipItem.index].totalWalletBTC && (
+            <Box>
+              <span className="label">BTC:</span>
+              <span>{formatFloat(list[tooltipItem.index].totalWalletUSDT)}</span>
```
Comment (cwagner22) [inline src/components/Balance/TotalEquity/TotalEquityGraph/TotalEquityGraph.js:76]: Should be `totalWalletBTC` right?

File: src/components/Navigation/Header/TopBalance/BalanceBox/BalanceBox.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ * @property {JSX.Element|String} [value2]
+ */
+
+/**
+ * Render a part of the equity bar
+ *
+ * @param {DefaultProps} props Component props.
+ * @returns {JSX.Element} JSX
+ */
+const BalaneBox = ({ name, value1, value2 }) => {
```
Comment (cwagner22) [inline src/components/Navigation/Header/TopBalance/BalanceBox/BalanceBox.js:17]: typo but not important..

Comment (cwagner22) [reply src/components/Navigation/Header/TopBalance/BalanceBox/BalanceBox.js:17]: Also it's hard to guess what is value1 and value2, in the future let's try to give the component param distinguishable names (or at least document what they represent)

File: src/i18n/translations/vn.yml [hunk 6]
```diff
@@ -33,11 +33,27 @@ recover.error: "Liên kết đã hết hạn! Xin vui lòng bắt đầu lại t
 testimonials.title: "Testimonials"
 testimonials.loadmore: "Load More"
 # BALANCE
-balance.available: "Khả dụng"
-balance.invested: "Đã Đầu Tư"
-balance.profit: "Lợi Nhuận Chưa Thực Hiện"
-balance.total: "Tổng vốn sở hữu"
-balance.profitlosses: "UP / L #"
+balance.available: "Available"
```
Comment (cwagner22) [inline src/i18n/translations/vn.yml:36]: Any reason the other translation files have been updated to contain english content?

File: src/store/store.js [hunk 7]
```diff
@@ -43,13 +43,14 @@ const migrations = {
...
+  17: (/** @type {PersistedDefaultState} */ state) => {
     return {
       ...state,
       settings: {
         ...state.settings,
         displayColumns: {
           ...state.settings.displayColumns,
+          futuresDailyBalance: [...initialState.settings.displayColumns.futuresDailyBalance],
           providerContracts: [...initialState.settings.displayColumns.providerContracts],
           providerOrders: [...initialState.settings.displayColumns.providerOrders],
```
Comment (cwagner22) [inline src/store/store.js:55]: FYI I don't think you have to destructure the array here as well

File: src/components/Navigation/Header/TopBalance/BalanceBox/BalanceBox.js (hunk 5 above)
Comment (muradmalik777) [reply src/components/Navigation/Header/TopBalance/BalanceBox/BalanceBox.js:17]: yeah, I will refactor these in the future and document them for now.

File: src/components/Balance/TotalEquity/TotalEquityGraph/TotalEquityGraph.js (hunk 3 above)
Comment (muradmalik777) [reply src/components/Balance/TotalEquity/TotalEquityGraph/TotalEquityGraph.js:46]: sure

File: src/components/Balance/TotalEquity/TotalEquityGraph/TotalEquityGraph.js (hunk 4 above)
Comment (muradmalik777) [reply src/components/Balance/TotalEquity/TotalEquityGraph/TotalEquityGraph.js:76]: yeah, my mistake.

File: src/components/Balance/ProfitLossAnalysis/ProfitLossAnalysis.js (hunk 2 above)
Comment (muradmalik777) [reply src/components/Balance/ProfitLossAnalysis/ProfitLossAnalysis.js:36]: sure yeah.

File: src/components/Balance/History/FuturesHistoryTable.js (hunk 1 above)
Comment (muradmalik777) [reply src/components/Balance/History/FuturesHistoryTable.js:116]: yes we can, this is just what I follow from school :D

File: src/i18n/translations/vn.yml (hunk 6 above)
Comment (muradmalik777) [reply src/i18n/translations/vn.yml:36]: well let me remember,  actually this was missing from here, so I just added here.

File: src/store/store.js (hunk 7 above)
Comment (muradmalik777) [reply src/store/store.js:55]: I think we should, otherwise it will just assign that reference, no?

Comment (muradmalik777) [reply src/store/store.js:55]: Done

File: src/components/Balance/History/FuturesHistoryTable.js (hunk 1 above)
Comment (cwagner22) [reply src/components/Balance/History/FuturesHistoryTable.js]: `i` is usually a coding standard.

**Pull request merged**